### PR TITLE
Fix Crash on app open and on tab close

### LINKED ISSUE #289: Quit web page error
**Describe the issue** 
```Lorsqu'on quitte une page internet ouverte dans l'appli (ex.: Moodle), l'application se ferme et met un message que l'application a cessé de fonctionner. version Android. Merci!```

**Screenshot** 
screenshot

**Device Infos** 
- **Device:** SM-G970W 
- **Version:** 4.4.8 
- **Connectivity:** ConnectivityResult.wifi 
- **Build number:** 1634853217 
- **Platform operating system:** android 
- **Platform operating system version:** RP1A.200720.012.G970WVLS6GUJ1

### LINKED ISSUE #304: Crash repeatedly from chrome custom tab
**Describe the bug**
I don't know if this is prevented safely handled, but it seems that a lot of user can't open the app due to a crash occurring in the package: `flutter_inappwebview`. It would be important to investigate this issue, but I don't know if this is reproducible it seems like it is trigger at random.

<details>
  <summary>Error log</summary>
  <pre>
  <code>
<b>Fatal Exception: java.lang.RuntimeException Unable to start activity</b>     
ComponentInfo{ca.etsmtl.applets.etsmobile/com.pichillilorenzo.flutter_inappwebview.chrome_custom_tabs.ChromeCustomTabsActivity}: java.lang.NullPointerException:
Attempt to read from field 'com.pichillilorenzo.flutter_inappwebview.InAppWebViewFlutterPlugin com.pichillilorenzo.flutter_inappwebview.chrome_custom_tabs.ChromeSafariBrowserManager.plugin' on a null object reference
  </code>
  </pre>
</details>

**To Reproduce**
?

**Expected behavior**
N/A

**Screenshots**
image

**Smartphone (please complete the following information):**
image

**Additional context**
It seems like we are not alone with this mystery bug:
https://github.com/pichillilorenzo/flutter_inappwebview/issues/1031
https://github.com/pichillilorenzo/flutter_inappwebview/issues/1009
https://github.com/pichillilorenzo/flutter_inappwebview/issues/801

### PR DESCRIPTION
**Pull request by MysticFragilist**
[flutter_customs_tabs](https://pub.dev/packages/flutter_custom_tabs) is much more maintained and seems more up to date than the package we were using before. I even put an animation on the page load with it 😄 

I can't really fully test the new package because of the bug right now. But as soon as I will be able to, I will drop some screenshot here :) At least for now, it doesn't crash on launch should be a good sign!

Edit: There it is! It looks like that really close to the previous behavior :

closes #304 
closes #289

### CODE REVIEW

**Review by AlexandreMarkus — CHANGES_REQUESTED** [review]

File: lib/core/viewmodels/web_link_card_viewmodel.dart [hunk 1]
```diff
@@ -33,24 +31,39 @@ class WebLinkCardViewModel extends BaseViewModel {
...
+    await launch(
+      url,
+      customTabsOption: CustomTabsOption(
+        toolbarColor: AppTheme.etsLightRed,
+        enableDefaultShare: false,
+        enableUrlBarHiding: true,
+        showPageTitle: true,
+        animation: CustomTabsSystemAnimation.slideIn(),
+        extraCustomTabs: const <String>[
+          // ref. https://play.google.com/store/apps/details?id=org.mozilla.firefox
```
[the remark this answers is not in the document]
Comment (MysticFragilist) [reply lib/core/viewmodels/web_link_card_viewmodel.dart:53]: Yess good idea! I'll check that!

File: lib/core/viewmodels/web_link_card_viewmodel.dart [hunk 2]
```diff
@@ -33,24 +31,39 @@ class WebLinkCardViewModel extends BaseViewModel {
       try {
         await launchInBrowser(link.link);
       } catch (error) {
+        // An exception is thrown if browser app is not installed on Android device.
+        _analyticsService.logError(
+            "WebLinkCardViewModel", "Error while launching url in tabs $error");
```
[the remark this answers is not in the document]
Comment (MysticFragilist) [reply lib/core/viewmodels/web_link_card_viewmodel.dart:36]: True true forgot about that one! 😅  I'll remove that then

**Review by AlexandreMarkus — APPROVED** [review]
LGTM!! Hopefully the browser will be more stable than before!

**Comment by MysticFragilist** [comment]
> LGTM!! Hopefully the browser will be more stable than before!

I hope so too!

**Review by eneiss — APPROVED** [review]
LGTM 👌

**Pull request merged**